test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs

```diff
@@ -0,0 +1,285 @@
+                    var cts = new CancellationTokenSource();
+                    cts.CancelAfter(10000);
+                    string[] platformInfo = await Process.RunAsync("lsb_release", "-sir", cts.Token);
+                    if (platformInfo.Length == 1)
+                    {
+                        platformInfo = platformInfo[0].Split(' ');
+                    }
+
+                    string os = platformInfo[0].Trim();
+                    string version = platformInfo[1].Trim();
+                    SupportedPackageExtension packageExtension = SupportedPackageExtension.None;
+
+                    switch (os)
+                    {
+                        case "Ubuntu":
+                            os = "ubuntu";
+                            packageExtension = SupportedPackageExtension.Deb;
+                            break;
+                        case "Raspbian":
+                            os = "debian";
+                            version = "stretch";
+                            packageExtension = SupportedPackageExtension.Deb;
+                            break;
+                        case "CentOS":
+                            os = os.ToLower();
+                            version = version.Split('.')[0];
+                            packageExtension = SupportedPackageExtension.Rpm;
+
+                            if (version != "7")
+                            {
+                                throw new NotImplementedException($"Don't know how to install daemon on operating system '{os} {version}'");
+                            }
+
+                            break;
+                        default:
+                            throw new NotImplementedException($"Don't know how to install daemon on operating system '{os}'");
+                    }
+
+                    return new LinuxStandardBase(os, version, packageExtension);
+                });
+
+        string forceInstallConfigCmd = null;
+        public string ForceInstallConfigCmd
+        {
+            get
+            {
+                if (this.forceInstallConfigCmd == null)
+                {
+                    LinuxStandardBase lsb = this.GetLsbResult().Result;
+                    switch (lsb.PackageExtension)
+                    {
+                        case SupportedPackageExtension.Deb:
+                            this.forceInstallConfigCmd = "dpkg --force-confnew -i";
+                            break;
+                        case SupportedPackageExtension.Rpm:
+                            this.forceInstallConfigCmd = "rpm -iv --replacepkgs";
+                            break;
+                        default:
+                            throw new NotImplementedException($"Undefined forceInstallConfigCmd for '{lsb.PackageExtension}'");
+                    }
+                }
+
+                return this.forceInstallConfigCmd;
+            }
+        }
+
+        string installCmd = null;
+        public string InstallCmd
```

Comment:
There are 2 problems with these properties:
1. Properties can't be async so you're using `.Result` instead, which is what we were trying to avoid (mixing sync/async patterns).
2. Each property is a lot of overhead, considering what it does. It has to await the LSB result, then switch on the package extension to produce a string. But most of these properties are used in `GetInstallCommandsFromLocal` or `GetInstallCommandsFromMicrosoftProd`, which also await the LSB result and then switches on the package extension to produce a string. Why not get rid of these properties and insert the strings directly into the `GetInstallCommandsFrom*` methods?

Reply:
Sure!